src/app/ipv/middleware.js

```diff
@@ -441,6 +448,40 @@ module.exports = {
       next(error);
     }
   },
+  formRadioButtonChecked: async (req, res, next) => {
+    const allowedPathRule = {
+      pattern: /^\/ipv\/page\/[^/]+$/,
+      methods: ["GET", "POST"],
+    };
+
+    const isPathAllowed = (url, method) => {
+      const path = new URL(
+        url,
+        `https://${req.headers.host}`,
+      ).pathname.toLowerCase();
+      if (!method || allowedPathRule.methods.includes(method)) {
+        if (path.match(allowedPathRule.pattern)) {
+          return true;
+        }
+        throw new Error("Path is not allowed");
+      }
+      return false;
+    };
+    try {
+      if (
+        req.method === "POST" &&
+        req.body.journey === undefined &&
+        isPathAllowed(req.originalUrl, req.method)
+      ) {
+        const redirectUrl = `${req.baseUrl}${req.path}?errorState=true`;
+        return res.redirect(redirectUrl);
```

Comment:
It doesn't seem right to redirect the user to another URL - can't we just show the error message on the page we are going to render?